File: python/2nd_week/Praktikum2/latihanpraktikum2_python/Product.py
class Product:
    # implementasi class product

    # constructor
    def __init__(self):
        self.price = 0
        self.idProduct = ""
    
    # set and get method
    def setPrice(self, price):
        self.price = price
    
    def setIdProduct(self, idProduct):
        self.idProduct = idProduct
    
    def getPrice(self):
        return self.price
    
    def getIdProduct(self):
        return self.idProduct

File: cpp/2nd_Week/Praktikum2/latihanpraktikum2_cpp/Product.cpp
#include <iostream>
#include <string>
using namespace std;

// parent class
class Product{
    // implementasi class Product
    private:
        long price;             //atribut price dengan tipe data long
        string idProduct;       //atribut idProduct dengan tipe data string
    public:
        //constructor       
        Product(){}
        
        /* set dan get method dari atribut */
        void setPrice(long paramPrice){
            price = paramPrice;
        }

        void setIdProduct(string paramIdProduct){
            idProduct = paramIdProduct;
        }

        long getPrice(){
            return price;
        }

        string getIdProduct(){
            return idProduct;
        }

        ~Product(){}
};